packages/pymedphys_deliverydata/src/pymedphys_deliverydata/utilities/core.py

```diff
@@ -66,35 +59,25 @@ def extract_angle_from_delivery_data(delivery_data, gantry_angle,
     return mu, mlc, jaw
 
 
-def find_relevant_control_points(mu):
-    """Returns that control points that had an MU difference either side.
-    """
-    mu_diff = np.diff(mu)
-    no_change = mu_diff == 0
-    no_change_before = no_change[0:-1]
-    no_change_after = no_change[1::]
+def strip_delivery_data(delivery_data, skip_size):
+    DeliveryDataObject = type(delivery_data)
 
-    no_change_before_and_after = no_change_before & no_change_after
-    irrelevant_control_point = np.hstack(
-        [no_change[0], no_change_before_and_after, no_change[-1]])
-    relevant_control_points = np.invert(irrelevant_control_point)
+    new_delivery_data = []
+    for item in delivery_data:
+        new_delivery_data.append(np.array(item)[::skip_size].tolist())
 
-    return relevant_control_points
+    return DeliveryDataObject(*new_delivery_data)
 
 
-def remove_irrelevant_control_points(mu, mlc, jaw):
-    """Removes control points that don't have MU delivery
-    """
-    assert len(mu) > 0, "No control points found"
+def filter_out_irrelevant_control_points(delivery_data: _DeliveryDataBase) -> _DeliveryDataBase:
```

Comment:
This was **BORN** to be a *method* of `DeliveryData`! It causes a simple state change of a `DeliveryData` object. This is very ugly (to me) as a function...

Reply:
Thing is, it can run on anything that looks like a delivery data object... if it was a method only, that wouldn't be possible.